cpp/src/DataStorm/NodeSessionManager.cpp

```diff
@@ -444,11 +446,13 @@ NodeSessionManager::disconnected(const LookupPrx& lookup)
 }
 
 void
-NodeSessionManager::destroySession(const NodePrx& node)
+NodeSessionManager::destroySession(const ConnectionPtr& connection, const NodePrx& node)
 {
     unique_lock<mutex> lock(_mutex);
+    // If the session is still using the connection destroy it, otherwise the node has already
+    // replace its NodeSession and it is using a new connection.
```

Comment:
```suggestion
    // Destroy the connection if the session is still using it, otherwise the node has already
    // replaced its NodeSession and it is using a new connection.
```